include/mbgl/platform/settings.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <mapbox/value.hpp>
+
+#include <memory>
+
+namespace mbgl {
+namespace platform {
+
+#define MAPBOX_SETTING(name, value) constexpr const char* name = "mapbox_" #value
```

Comment:
DECLARE_MAPBOX_SETTING

Reply:
nit: `const char name[] = `

Reply:
@pozdnyakov why?

Reply:
to avoid having of an extra pointer :smile: 
https://www.geeksforgeeks.org/whats-difference-between-char-s-and-char-s-in-c/

Reply:
https://godbolt.org/z/hqc6-Z mostly the same assembly, and we use constexpr strings in all codebase.